packages/m2m-gateway/src/model/errors.ts

```diff
@@ -202,3 +204,23 @@ export function eserviceDescriptorInterfaceNotFound(
     title: "Eservice descriptor interface not found",
   });
 }
+
+export function agreementPurposeNotFound(
```

Comment:
Consistent with `getPurposeAgreement`
```suggestion
export function purposeAgreementNotFound(
```